### ISSUE TITLE
component is mounted even if breakpoint is false

### ISSUE THREAD

**Issue by MatteoGabriele** [e2o/vue-element-query#8]
Hi there! Not sure if you are still maintaining this package, but I found a bug while using it.

As you can see from the following example, the Test component should be rendered only when the Parent it's bigger than 500px.
Everything works fine on the surface, but there's a "small" issue: when `$eq.medium` is `false`, the Test component is immediately mounted and destroyed, triggering all the eventual logic inside of it.
I wanted to create a small reproduction on CodeSandbox, but this package doesn't seem to work in that environment for some reason.

```html
<template>
  <div id="app">
    <test v-if="$eq.medium" />
  </div>
</template>

<script>
import Test from "@/components/Test";

export default {
  eq: {
    breakpoints: {
      medium: { minWidth: 500 },
    },
  },

  components: {
    Test,
  },
};
</script>
```

Thanks again for your work!

**Comment by e2o** [comment]
Hello there,

I'm not actively maintaining this library, but I appreciate you reaching out to me so I will see how we can resolve this issue.

I just created a CSB describing your issue, and I don't run into this. https://codesandbox.io/s/b4zss
Perhaps you could share your Test-component to see how the logic inside there is triggering?

**Comment by MatteoGabriele** [comment]
Thanks a lot for your time.
Indeed your example does work, but once I've tried a v-if/v-else combo, then the issue was "finally" there.
I've used your CSB as a reference and made some changes.

As you can see Component2 is mounted and destroyed

https://codesandbox.io/s/laughing-wind-nj5n0?file=/src/App.vue

**Comment by MatteoGabriele** [comment]
you can get the bug also with your example: just replace `activated` with `mounted` and invert your if statement in the template

```html
<test v-if="!$eq.medium" />
```

**Comment by e2o** [comment]
Okay, i localized the issue and here is what is happening.
As soon as the plugin is "ready" to observe any component-sizes, it returns an object (even empty) as value for `$eq`.

This means that naturally;
```
<component1 v-if="$eq.medium" />
<component2 v-else />
```

Will always render one or the other. As `$eq` will always be an object, and `$eq.medium` will either be `undefined`, `false` or `true`.

To circumvent this, with the changes I made to the plugin you could do the following to have your scenario work as intended:
```
<template v-if="$eq">
  <component1 v-if="$eq.medium" />
  <component2 v-else />
</template>
```

As a downside, all previous instances where this check is not present would break as `$eq` will be null, and trying to access `$eq.medium` would result in an errow being thrown.

Any ideas on how I should move forward with this? Ideally I don't want force users to update their components to check for `$eq` first. But the option to have strict rendering should be there, as to serve users dealing with your usecase.

If you are curious, I forked the plugin into CSB to play around with:  https://codesandbox.io/s/vigilant-tereshkova-1fccs
I only made 2 small adjustments (just search for `// adjustment` in the `plugin.js`)

It's been a while (2-3 years) since I worked with Vue, so i'm a bit rusty and I might be overlooking an easier solution for all this.

Kindly awaiting your input,
Enzo

**Comment by MatteoGabriele** [comment]
Yeah, I see it now. It makes sense that the component is getting momentaneously rendered and destroyed.

You obviously can't publish the `$eq` as `null` without a major release. 

The solution with the least amount of work would be to provide a property inside `$eq` that tells us if the plugin is ready.

```html
<div v-if="$eq.isObserving">
  <div v-if="$eq.medium">only on medium</div>
  <div v-else>only on small</div>
</div>
```

We could also provide a directive or a component to wrap elements so that all checks are done internally, although I'm not sure how I feel about it.

```html
<div v-breakpoint="medium">only on medium</div>
<div v-breakpoint.not="medium">everything else</div>

<eq breakpoint="medium">only on medium</eq>
<!-- mandatory to create a small breakpoint -->
<eq breakpoint="small">only on small</eq>
```

I'm going to play with it a bit and come back to you.

Thanks a lot for your help

**Comment by e2o** [comment]
Thanks again for your input on this issue 👏 

Your PR was included in [v3.1.0](https://github.com/e2o/vue-element-query/releases/tag/v3.1.0).

**Comment by MatteoGabriele** [comment]
Thanks, man! 🙏 
Cheers!
